### PR TITLE
Fix notify function when no type is passed

### LINKED ISSUE #6767: Can't call notify function without `type` argument
**What you were expecting:**
Be able to not pass optional argument `type` to `useNotify` returned function

**What happened instead:**
Code breaks because destructuring `undefined` throws an error.

**Steps to reproduce:**
```
const notify = useNotify();
notify();
```

**Environment**

* React-admin version: 3.19.0 and I think will happen on 3.19.1 too.

### PR DESCRIPTION
**Pull request by nidebo**
#6713 was fixing this problem but #6760 makes it happen again

Close #6767

### CODE REVIEW

**Comment by WiXSL** [comment]
I've had added the exact change in #6760, but reverted for some reason, It seams ok.

File: packages/ra-core/src/sideEffect/useNotify.ts [hunk 1]
```diff
@@ -52,9 +52,11 @@ const useNotify = () => {
                         }
                     )
                 );
-            } else {
+            } else if (type) {
                 const { type: messageType, ...options } = type;
                 dispatch(showNotification(message, messageType, options));
+            } else {
+                dispatch(showNotification(message));
```
Comment (djhi) [inline packages/ra-core/src/sideEffect/useNotify.ts:59]: Thanks!

```suggestion
            } else {
                const { type: messageType, ...options } = type || {};
                dispatch(showNotification(message, messageType, options));
            }
```

**Review by djhi — CHANGES_REQUESTED** [review]

File: packages/ra-core/src/sideEffect/useNotify.ts (hunk 1 above)
Comment (nidebo) [reply packages/ra-core/src/sideEffect/useNotify.ts]: Applied, thanks!

**Review by djhi — APPROVED** [review]
Thanks!

**Pull request merged**